content/doc/book/system-administration/reverse-proxy-configuration-with-jenkins.adoc

```diff
@@ -0,0 +1,1411 @@
+back to the client. A typical set up for mod_proxy would look like this:
+
+[source]
+----
+upstream jenkins {
+  keepalive 32; # keepalive connections
+  server 127.0.0.1:8080; # jenkins ip and port
+}
+
+server {
+  listen          80;       # Listen on port 80 for IPv4 requests
+
+  server_name     jenkins.example.com;
+
+  # this is the jenkins web root directory
+  # (mentioned in the /etc/default/jenkins file)
+  root            /var/run/jenkins/war/;
+
+  access_log      /var/log/nginx/jenkins/access.log;
+  error_log       /var/log/nginx/jenkins/error.log;
+  # pass through headers from Jenkins that Nginx considers invalid
+  ignore_invalid_headers off;
+
+  location ~ "^/static/[0-9a-fA-F]{8}\/(.*)$" {
+    #rewrite all static files into requests to the root
+    #E.g /static/12345678/css/something.css will become /css/something.css
+    rewrite "^/static/[0-9a-fA-F]{8}\/(.*)" /$1 last;
+  }
+
+  location /userContent {
+    # have nginx handle all the static requests to userContent folder
+    #note : This is the $JENKINS_HOME dir
+    root /var/lib/jenkins/;
+    if (!-f $request_filename){
+      #this file does not exist, might be a directory or a /**view** url
+      rewrite (.*) /$1 last;
+      break;
+    }
+    sendfile on;
+  }
+
+  location / {
+      sendfile off;
+      proxy_pass         http://jenkins;
+      proxy_redirect     default;
+      proxy_http_version 1.1;
+
+      proxy_set_header   Host              $host;
+      proxy_set_header   X-Real-IP         $remote_addr;
+      proxy_set_header   X-Forwarded-For   $proxy_add_x_forwarded_for;
+      proxy_set_header   X-Forwarded-Proto $scheme;
+      proxy_max_temp_file_size 0;
+
+      #this is the maximum upload size
+      client_max_body_size       10m;
+      client_body_buffer_size    128k;
+
+      proxy_connect_timeout      90;
+      proxy_send_timeout         90;
+      proxy_read_timeout         90;
+      proxy_buffering            off;
+      proxy_request_buffering    off; # Required for HTTP CLI commands
+      proxy_set_header Connection ""; # Clear for keepalive
+  }
+
+}
+----
+
+This assumes that you run Jenkins on port 8080.
+Remember to create the folder /var/log/nginx/jenkins.
+
+For this set up to work, the context path of Jenkins must be the same
+between your Nginx and Jenkins (that is, you can't run Jenkins on
+http://localhost:8081/ci and have it exposed at
+http://localhost:80/jenkins).
+
+Set the context path by modifying the hudson.xml configuration file and
+adding --prefix=/hudson (or similar) to the <arguments> entry.
+
+If you are having problems with some paths (eg folders) with *Blue
+Ocean*, you may need to add the following snippet to your proxy
+configuration:
+
+[source]
+----
+if ($request_uri ~* "/blue(/.*)") {
+    proxy_pass http://YOUR_SERVER_IP:YOUR_JENKINS_PORT/blue$1;
+    break;
+}
+----
+
+To give nginx permission to read Jenkins web root folder, add nginx to
```

Comment:
The Nginx project prefers that the name start with a captial 'N' as far as I can tell.
```suggestion
To give Nginx permission to read Jenkins web root folder, add the `nginx` user to
```